src/entities/activity/ui/items/ActionPlan/Phrase.tsx

```diff
@@ -1,63 +1,98 @@
+  phraseId,
+  documentData,
+  pageComponents,
+  isFirstOnPage,
+  isLastOnPage,
+}: PhraseProps) => {
   const gap = useXScaledDimension(24);
   const minHeight = useXScaledDimension(72);
   const imageWidth = useXScaledDimension(67);
   const imageHeight = useXScaledDimension(66);
   const imagePadding = useXScaledDimension(2);
   const fontSize = useXScaledDimension(16);
   const lineHeight = useYScaledDimension(24);
-  const correlatedPageMaxHeightLineCount = useCorrelatedPageMaxHeightLineCount();
-  const maxLineCount = correlatedPageMaxHeightLineCount.lineCount;
 
-  const { components } = phrase.fields.reduce(
-    (acc, field, fieldIndex) => {
-      const isLineStart = fieldIndex === 0 || acc.prevField?.type === 'line_break';
+  const phraseComponents = useMemo(
+    () =>
+      pageComponents.reduce((acc, component, componentIndex) => {
+        if (component.phraseId === phraseId) {
+          if (
+            (isFirstOnPage && componentIndex === 0) ||
+            (isLastOnPage && componentIndex === pageComponents.length - 1)
+          ) {
+            // Remove leading and trailing line-break components.
+            if (component.componentType !== 'line_break') {
+              acc.push(component);
+            }
+          } else {
+            acc.push(component);
+          }
+        }
+        return acc;
+      }, [] as PageComponent[]),
+    [pageComponents, phraseId, isFirstOnPage, isLastOnPage],
+  );
+
+  const renderedComponents = useMemo(() => {
+    const rendered: React.ReactNode[] = [];
+
+    let previousComponentType: PageComponent['componentType'] | undefined;
+    phraseComponents.forEach((component, componentIndex) => {
+      const componentType = component.componentType;
+      const isAtStart = componentIndex === 0 || previousComponentType === 'line_break';
 
-      if (field.type === 'sentence') {
-        acc.components.push(<TextSegment text={field.text} isAtStart={isLineStart} />);
-      } else if (field.type === 'item_response') {
-        acc.components.push(
-          <ResponseSegment phrasalData={phrasalData} field={field} isAtStart={isLineStart} />,
-        );
-      } else if (field.type === 'line_break') {
-        acc.components.push(<br />);
+      if (componentType === 'sentence') {
+        rendered.push(<TextSegment text={component.text} isAtStart={isAtStart} />);
+      } else if (componentType === 'item_response') {
+        let itemResponse: ComponentProps<typeof ResponseSegment>['itemResponse'];
+        if (component.itemResponseType === 'list') {
+          itemResponse = {
+            itemResponseType: 'list',
+            items: component.items,
+          };
+        } else {
+          itemResponse = {
+            itemResponseType: 'text',
+            text: component.text,
+          };
+        }
+        rendered.push(<ResponseSegment itemResponse={itemResponse} isAtStart={isAtStart} />);
+      } else if (componentType === 'line_break') {
+        rendered.push(<br />);
```

Comment:
This came up in a discussion between @farmerpaul and I, for a [phrase-builder-related PR in the Admin app](https://github.com/ChildMindInstitute/mindlogger-admin/pull/1958#discussion_r1816667345):

In Figma, the specs mention that the "line break" item should introduce [a 32px high gap](https://www.figma.com/design/3wcZ6bxK2QVo733ADxVjBw/SSI-2024?node-id=692-76427&t=BLw76cfZQzfaJAK6-4), which a `<br />` doesn't do. There also isn't a great cross-browser way of styling a `<br />` to include such a gap.

In my PR on the admin app, I render line breaks like so:

```jsx
<Box component="hr" sx={{ m: 0, height: 32, border: 'none' }} />
```

… which you could theoretically just drop in right here for the same effect.

This is a large PR that I haven't been able to look at in detail, so hopefully this doesn't have an adverse effect 😅.

Reply:
I don't think it matter exactly what gets rendered, as the height calculation should just take whatever that gets rendered into account. So I should be able to swap this out.